### PR TITLE
Delegate username resolver - Closes #181

### LINKED ISSUE #181: Delegate username resolver
### Description

Add missing feature that resolved AccountID/public key to username.

### Motivation

- Account ID/Username resolution is one of the key features of Lisk Service
- This change has to be made in order to avoid breaking changes in the API

### Acceptance Criteria

- Account and delegate endpoints report correct usernames

### PR DESCRIPTION
**Pull request by nagdahimanshu**
### What was the problem?
This PR resolves #181

### How was it solved?
- [x] Account endpoint reports correct usernames
- [x] Delegate endpoints reports correct usernames

### How was it tested?
Locally

### CODE REVIEW

File: services/gateway/sources/mappings/account.js [hunk 1]
```diff
@@ -27,6 +28,12 @@ module.exports = {
 		rewards: '=,string',
 		username: '=,string',
 		vote: '=,string',
+		totalVotesReceived: ',string',
+		isBanned: '=',
+		status: '=,string',
+		pomHeights: '=',
```
Comment (sameersubudhi) [inline services/gateway/sources/mappings/account.js:34]: For a consistent experience, should `pomHeights` be restructured as an array of objects containing `start` and `end` properties just like in delegates? Refer to the structure definition below:
```
	pomHeights: ['pomHeights', {
		start: '=,string',
		end: '=,string',
	}],
```

File: services/core/shared/core/compat/sdk_v4/accounts.js [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ *
+ */
+const coreApi = require('./coreApi');
+const coreCache = require('../sdk_v2');
```
Comment (sameersubudhi) [inline services/core/shared/core/compat/sdk_v4/accounts.js:17]:
```suggestion
const {
	getCachedAccountByAddress,
	getCachedAccountByPublicKey,
	getCachedAccountBySecondPublicKey,
} = require('../sdk_v2');
```

File: services/core/shared/core/compat/sdk_v4/accounts.js [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+const parseAddress = address => {
+	if (typeof address !== 'string') return '';
+	return address.toUpperCase();
+};
+
+const validatePublicKey = publicKey => (typeof publicKey === 'string' && publicKey.match(/^([A-Fa-f0-9]{2}){32}$/g));
+
+const confirmAddress = async address => {
+	if (!address || typeof address !== 'string') return false;
+	const account = await coreCache.getCachedAccountByAddress(parseAddress(address));
```
Comment (sameersubudhi) [inline services/core/shared/core/compat/sdk_v4/accounts.js:28]:
```suggestion
	const account = await getCachedAccountByAddress(parseAddress(address));
```

File: services/core/shared/core/compat/sdk_v4/accounts.js [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+	return (account && account.address === address);
+};
+
+const confirmPublicKey = async publicKey => {
+	if (!publicKey || typeof publicKey !== 'string') return false;
+	const account = await coreCache.getCachedAccountByPublicKey(publicKey);
```
Comment (sameersubudhi) [inline services/core/shared/core/compat/sdk_v4/accounts.js:34]:
```suggestion
	const account = await getCachedAccountByPublicKey(publicKey);
```

File: services/core/shared/core/compat/sdk_v4/accounts.js [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+	return (account && account.publicKey === publicKey);
+};
+
+const confirmSecondPublicKey = async secondPublicKey => {
+	if (!secondPublicKey || typeof secondPublicKey !== 'string') return false;
+	const account = await coreCache.getCachedAccountBySecondPublicKey(secondPublicKey);
```
Comment (sameersubudhi) [inline services/core/shared/core/compat/sdk_v4/accounts.js:40]:
```suggestion
	const account = await getCachedAccountBySecondPublicKey(secondPublicKey);
```

**Review by sameersubudhi — CHANGES_REQUESTED** [review]
Please find my review suggestions.

File: services/gateway/sources/mappings/account.js (hunk 1 above)
Comment (nagdahimanshu) [reply services/gateway/sources/mappings/account.js]: Updated

**Review by sameersubudhi — APPROVED** [review]
LGTM! 👍

**Review by MichalTuleja — CHANGES_REQUESTED** [review]
Please fix formatting.

**Review by MichalTuleja — APPROVED** [review]

**Pull request merged**